ArcGISRuntimeSDKQt_CppSamples/Geometry/ClipGeometry/ClipGeometry.cpp

```diff
@@ -0,0 +1,128 @@
+
+  // Set map to map view and append the two initial graphics overlays
+  m_mapView->setMap(m_map);
+  m_mapView->graphicsOverlays()->append(m_coloradoOverlay);
+  m_mapView->graphicsOverlays()->append(m_envelopesOverlay);
+
+  // Set initialViewpoint
+  Viewpoint initialViewpoint(Envelope(-12500000, 4000000, -11000000, 5500000, SpatialReference(3857)));
+  m_mapView->setViewpoint(initialViewpoint);
+
+  // Create the graphics
+  createGraphics();
+
+}
+
+// Creates the visual elements and adds them to their respective overlays
+void ClipGeometry::createGraphics(){
+
+  // Colorado outline + fill
+  m_coloradoOutline = new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, QColor("#FF0000FF"), 2.0f, this);
+  m_coloradoFill = new SimpleFillSymbol(SimpleFillSymbolStyle::Solid, QColor("#220000FF"), m_coloradoOutline, this);
+
+  // Create and add the Colorado graphic to its overlay
+  m_coloradoGraphic = new Graphic(m_coloradoEnvelope, m_coloradoFill, this);
+  m_coloradoOverlay->graphics()->append(m_coloradoGraphic);
+
+  // Clipping envelopes outline
+  SimpleLineSymbol* envelopeOutline = new SimpleLineSymbol(SimpleLineSymbolStyle::Dash, QColor("#FFFF0000"), 3.0f, this);
+
+  // Create and add the clipping envelope graphics to their overlay
+  m_outsideGraphic = new Graphic(m_outsideEnvelope, envelopeOutline, this);
+  m_intersectingGraphic = new Graphic(m_intersectingEnvelope, envelopeOutline, this);
+  m_insideGraphic = new Graphic(m_insideEnvelope, envelopeOutline, this);
+  m_envelopesOverlay->graphics()->append(m_outsideGraphic);
+  m_envelopesOverlay->graphics()->append(m_intersectingGraphic);
+  m_envelopesOverlay->graphics()->append(m_insideGraphic);
+
+}
+
+// Clips the overlapping geometries and displays them on the map
+void ClipGeometry::clipAreas(){
+
+  // Immediately hide the reference graphic
+  m_coloradoGraphic->setVisible(false);
+
+  // Get the graphicsListModel from the overlay containing the clipping envelopes
+  GraphicListModel* graphics = m_envelopesOverlay->graphics();
+
+  /* Iterate through the envelope graphics and call the static 'clip' function, passing in
+   * the reference graphic and the current iteration's envelope geometry as arguments.
+   * If the return value is valid, create a new graphic using the returned geometry and the
+   * fillSymbol of the reference graphic as arguments. Append the new graphic to its respective
+   * graphicsOverlay. */
+  for(int i = 0; i < graphics->size(); i++){
+    m_clippedGeometry = GeometryEngine::clip(m_coloradoGraphic->geometry(), graphics->at(i)->geometry().extent());
+    if(!m_clippedGeometry.isEmpty()){
+      Graphic* clippedGraphic = new Graphic(m_clippedGeometry, m_coloradoFill, this);
+      m_clippedAreasOverlay->graphics()->append(clippedGraphic);
+    }
+  }
+  // Append the graphicsOverlay to the mapView to render
+  m_mapView->graphicsOverlays()->append(m_clippedAreasOverlay);
```

Comment:
this should probably get moved to somewhere that we know it would only get called once (around line 100). We don't need to try and re append this overlay every time the function is called. After the first time, this will not do anything.

Reply:
Line 100 is outside of the scope of the function, do you mean anywhere near the top of the function (104?)

Reply:
nope, I mean it can be inside of componentCompleted. That way you know it'll ever only be called once. This function is a public Q_INVOKABLE and could (in theory) be called several times from QML. I understand that in your QML code you disable the button, but it would be better to add the overlay to the map view next to where you add the other overlays